nixos/modules/services/system/self-deploy.nix

```diff
@@ -0,0 +1,167 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.self-deploy;
+
+  workingDirectory = "/var/lib/self-deploy";
```

Comment:
Maybe ``/var/lib/nixos-self-deploy`` would be a better directory. Not sure though.

Reply:
Hmm. I doubt it will be necessary, but it might be a good idea to do anyway. Namespacing it with a `nixos-` prefix will all but ensure there's no chance of a collision.